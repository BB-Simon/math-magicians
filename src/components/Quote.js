import { useEffect, useState } from 'react';

const apiKey = 'RZq11UUitBvJnfp97sJacg==deXkuPUuPtcMYcEQ';
const category = 'funny';
const url = `https://api.api-ninjas.com/v1/quotes?category=${category}`;

const Quote = () => {
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);
  const [quotes, setQuotes] = useState([]);

  useEffect(() => {
    (async () => {
      try {
        const res = await fetch(url, {
          headers: {
            'X-Api-Key': apiKey,
          },
        });
        setQuotes(await res.json());
        setLoading(false);
      } catch (error) {
        setError(error.message);
        setLoading(false);
      }
    })();
  }, []);

  return (
    <div className="quote-container">
      {loading ? <p>Quote is loading...!</p> : null}
      {error ? <p>Error oqurred while fetching quote, please reload the page agin:(</p> : null}
      {!loading && !error ? (
        <div className="quotes-wrapper">
          <h1>Quotes</h1>
          <p>{`${quotes[0]?.quote} __ by ${quotes[0]?.author}`}</p>
        </div>
      ) : null}
    </div>
  );
};

export default Quote;
