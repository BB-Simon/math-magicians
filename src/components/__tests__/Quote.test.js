import TestRenderer from 'react-test-renderer';
import * as React from 'react';
import Quote from '../Quote';

describe('Jest Snapshot testing suite', () => {
  it('renders correctly', () => {
    const domTree = TestRenderer.create(<Quote />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
