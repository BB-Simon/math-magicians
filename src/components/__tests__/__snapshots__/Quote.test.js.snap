// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite renders correctly 1`] = `
<div
  className="quote-container"
>
  <p>
    Quote is loading...!
  </p>
</div>
`;
